### DISCUSSION TITLE
Different people with the webpack5 cache has different condition

### DISCUSSION THREAD

**Discussion by lanmch** [webpack/webpack#15072] [Q&A]
Last week, I add some data buried point about the ‘not hit cache module num’.
and I find some only have 0-100 modules not hit cache(need to build), so it build fast.
but some have more than 3k modules not hit cache.
(In our project, there are about 25000 modules all)
How can I fix it to improve the hit rate ?

**Comment by alexander-akait** [comment]
Maybe you have plugin/loader which invalidate cache? Because in all built-in plugins/loader cache works fine

**Reply by lanmch** [reply]
but all people have the same config with the same project, i dont think the invalidate cache will interfere someone but its ok with others

**Reply by alexander-akait** [reply]
How you understand which modules were invalidated? What is cache setup? How you debug it? Ideally you should not have invalidation at all.

**Reply by lanmch** [reply]
sorry, what i want to express is invalid invalid config not invalid cache

**Reply by alexander-akait** [reply]
Can you clarify, I don't understand you

**Reply by lanmch** [reply]
Ok, let me show my opinion
First, the phenomenon is different people run the same project with the same config cost different times, maybe some about 60s, some about 100s, and others more than 200s.
I try to add some data buried about the count of the modules witch not hit the cache, and i found 60s just about 10 modules need to be build, 100s about 1000 module need to be build,  200s+ about  more than 3000 module need to be build.(need to be build means  which not hit the cache).
At first I think(as you told me below), may be some invalidate config(like loaders or plugins) cause this difference problem. But after I think carefully, I think it shouldn't have different result with the same config?

**Reply by alexander-akait** [reply]
> First, the phenomenon is different people run the same project with the same config cost different times, maybe some about 60s, some about 100s, and others more than 200s.

Because different people have different computers with different file systems/os/hardware/etc

> At first I think(as you told me below), may be some invalidate config(like loaders or plugins) cause this difference problem. But after I think carefully, I think it shouldn't have different result with the same config?

If cache logic implemented wrong in a plugin, it can be a problem

If you need an exact answer why this is happening, you will need to provide an example, or at least webpack.config.js (or configuration which used)

**Reply by lanmch** [reply]
our config is too many... I showed it to you before, I don’t know if you still have an impression.
maybe is there some way to check this problem? is there any way to improve the hit rate?

**Reply by lanmch** [reply]
> Because different people have different computers with different file systems/os/hardware/etc

I think it has little to do with the difference in computer hardware configuration. Most people in my team use the macbook pro 16 2019/2.6 GHz/32G. 
I even counted everyone's boot time and memory usage. The data shows that they do not have a particularly strong association.

**Reply by alexander-akait** [reply]
> maybe is there some way to check this problem?

I show how to debug cache before, you can provide stats output here and I try to analyze it

> is there any way to improve the hit rate?

No way to improve the hit rate, module can be cached or not, they can be dropped from memory after some time (if you setup it) to reduce memory usage and no more

Please provide more information, the questions are very vague, so I cannot give any answers.

What is initial time? What is initial cache debug output? What is next compilation time and output? When you faced with bad caching on first/second/third/etc run? What is cache used - file system/memory? Which options you set for cache? Can you show output with cache? Do you have `uncachable` in output?

**Reply by lanmch** [reply]
hi, do i need delete  the temp_cache dir manually? I find that when the hit rate is low, i can delete the temp_Cache dir, then the cache works well and the hit rate has been improved

**Reply by alexander-akait** [reply]
What is `temp_cache`?

**Reply by lanmch** [reply]
> What is `temp_cache`?

the cache dir with webpack5

**Reply by alexander-akait** [reply]
You should not remove cache directory between builds, otherwise webpack will create cache files each run

**Reply by lanmch** [reply]
> You should not remove cache directory between builds, otherwise webpack will create cache files each run

but when I remove it, the project build with new cache directory become faster

**Reply by alexander-akait** [reply]
Again, if you need help and answers, please provide example how we can reproduce the problem, otherwise I can't help you, deleting cache file should be always slow, if you build is faster when you removed cache file, then most likely your cache simply does not work, it can be due non official plugins/loaders or wrong configuration
